GraphWebApi/Controllers/PermissionsController.cs

```
@@ -39,53 +42,78 @@ public PermissionsController(IPermissionsStore permissionsStore, TelemetryClient
         // Gets the permissions scopes
         [HttpGet]
         [Produces("application/json")]
-        public async Task<IActionResult> GetPermissionScopes([FromQuery]string scopeType = "DelegatedWork",
+        public async Task<IActionResult> GetPermissionScopes([FromQuery]ScopeType? scopeType = null,
                                                              [FromQuery]string requestUrl = null,
                                                              [FromQuery]string method = null,
                                                              [FromQuery]string org = null,
-                                                             [FromQuery]string branchName = null)
+                                                             [FromQuery]string branchName = null,
+                                                             [FromQuery]bool includeHidden = false,
+                                                             [FromQuery]bool isLeastPrivilege = false)
         {
-            string localeCode = RequestHelper.GetPreferredLocaleLanguage(Request) ?? Constants.DefaultLocale;
-            _telemetryClient?.TrackTrace($"Request to fetch permissions for locale '{localeCode}'",
-                                            SeverityLevel.Information,
-                                            _permissionsTraceProperties);
+            if (!string.IsNullOrEmpty(requestUrl) && string.IsNullOrEmpty(method))
```

What if the `requestUrl` is empty? Do we want to throw an error?

---

If `requestUrl` is null or empty then we return all permissions. We won't need to throw an error.